src/aten/sycl/UnaryKernels.cpp

```diff
@@ -18,6 +18,112 @@ struct AbsFunctor {
   }
 };
 
+template <typename scalar_t>
+struct SinFunctor {
+  scalar_t operator()(const scalar_t a) const {
+    return std::sin(a);
+  }
+};
+
+template <typename scalar_t>
+struct CosFunctor {
+  scalar_t operator()(const scalar_t a) const {
+    return std::cos(a);
+  }
+};
+
+template <typename scalar_t>
+struct LogFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return std::log(a);
+  }
+};
+
+template <typename scalar_t>
+struct SqrtFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return std::sqrt(a);
+  }
+};
+
+template <typename scalar_t>
+struct RsqrtFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return sycl::rsqrt(float(a));
+  }
+};
+
+template <>
+struct RsqrtFunctor<double> {
+  double operator()(double a) const {
+    return sycl::rsqrt(a);
+  }
+};
+
+template <typename T>
+struct RsqrtFunctor<c10::complex<T>> {
+  c10::complex<T> operator()(c10::complex<T> a) const {
+    return c10::complex<T>(1.0, 0) /
+        static_cast<c10::complex<T>>(
+               std::sqrt(static_cast<std::complex<T>>(a)));
+  }
+};
+
+template <typename scalar_t>
+struct TanhFunctor {
+  scalar_t operator()(scalar_t a) const {
+    using opmath_t = at::opmath_type<scalar_t>;
+    return std::tanh(static_cast<opmath_t>(a));
+  }
+};
+
+template <typename scalar_t>
+struct NegFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return -a;
+  }
+};
+
+template <typename scalar_t>
```

Comment:
Comment:
// For these common numeric helpers, we should share with CUDA code when putting XPU in-tree. `aten/src/ATen/native/cuda/UnaryFractionKernels.cu :69`